Improve error handling of AbortExecutionException in smallrye-graphql extension

### Description

While playing around with `quarkus`, `smallrye-graphql` and `spock` I was running into problems while trying to execute a [interaction based test case](https://spockframework.org/spock/docs/1.0/interaction_based_testing.html) for a graphql endpoint with a mocked bean (Spock mock)

```groovy
@PackageScope
@Inject
@GraphQLClient
DynamicGraphQLClient client

def "Assert that ......."() {
    given: "Mocked bean"
    QuarkusMock.installMockForType(Mock(Bean) {
        1 * get(_, "test",1) >> UUID.randomUUID()
        0 * _
    },Bean.class)
    when: "Action is performed that invokes a method that causes a org.spockframework.mock.TooManyInvocationsError (0 * _)"
    def response = client.executeSync(.....)
    then: "Test case fails with:  Unexpected response. Code=500, message=\"Internal Server Error\", body=\"null\""
    noExceptionThrown()
}
```

The test case always failed with an generic error message, giving me a hard time to detect the real cause of the failing test:

```
Expected no exception to be thrown, but got 'io.smallrye.graphql.client.InvalidResponseException'

	at spock.lang.Specification.noExceptionThrown(Specification.java:118)
	at xxxxxxxxxxxxxxxxxxx.Assert that ..................................(xxxxxxxxxxxxxxxxxxxxxxxxxxxx.groovy:558)
Caused by: io.smallrye.graphql.client.InvalidResponseException: Unexpected response. Code=500, message="Internal Server Error", body="null"
	at io.smallrye.graphql.client.impl.ResponseReader.readFrom(ResponseReader.java:64)
	at io.smallrye.graphql.client.vertx.dynamic.VertxDynamicGraphQLClient.lambda$executeSingleResultOperationOverHttp$7(VertxDynamicGraphQLClient.java:338)
	at io.smallrye.context.impl.wrappers.SlowContextualFunction.apply(SlowContextualFunction.java:21)
	at io.smallrye.mutiny.operators.uni.UniOnItemTransform$UniOnItemTransformProcessor.onItem(UniOnItemTransform.java:36)
	at io.smallrye.mutiny.operators.uni.builders.UniCreateFromCompletionStage$CompletionStageUniSubscription.forwardResult(UniCreateFromCompletionStage.java:63)
	at io.smallrye.context.impl.wrappers.SlowContextualBiConsumer.accept(SlowContextualBiConsumer.java:21)
```


After some debugging I found out that the exception is thrown in [smallrye-graphql](https://github.com/smallrye/smallrye-graphql/blob/9cefecb754da67af0da8d2ed714bb7e88878baf1/server/implementation/src/main/java/io/smallrye/graphql/execution/datafetcher/helper/ReflectionInvoker.java#L110) and then catched by the quarkus graphql extension in https://github.com/quarkusio/quarkus/blob/6014d34911cd0873487994ebdafc7c883972a676/extensions/smallrye-graphql/runtime/src/main/java/io/quarkus/smallrye/graphql/runtime/spi/datafetcher/QuarkusDefaultDataFetcher.java#L89-L92

and finally handled in https://github.com/quarkusio/quarkus/blob/6014d34911cd0873487994ebdafc7c883972a676/extensions/smallrye-graphql/runtime/src/main/java/io/quarkus/smallrye/graphql/runtime/SmallRyeGraphQLExecutionHandler.java#L324-L327


So the exception/cause of the problem gets swallowed ->  No log message

Since such errors are bypassing all ExceptionHandler, ErroCodes .... mechanism, it's also not possible to make such exceptions appear in the logs by customization


### Implementation ideas

A simple log entry would help a lot.

--- a/extensions/smallrye-graphql/runtime/src/main/java/io/quarkus/smallrye/graphql/runtime/SmallRyeGraphQLExecutionHandler.java
+++ b/extensions/smallrye-graphql/runtime/src/main/java/io/quarkus/smallrye/graphql/runtime/SmallRyeGraphQLExecutionHandler.java
@@ -13,9 +13,12 @@ import javax.json.JsonObject;
 import javax.json.JsonObjectBuilder;
 import javax.json.JsonReader;
 
+import org.jboss.logging.Logger;
+
 import graphql.ErrorType;
 import graphql.ExecutionResult;
 import graphql.GraphQLError;
+import graphql.execution.AbortExecutionException;
 import io.quarkus.security.identity.CurrentIdentityAssociation;
 import io.quarkus.vertx.http.runtime.CurrentVertxRequest;
 import io.smallrye.graphql.execution.ExecutionResponse;
@@ -45,6 +48,8 @@ public class SmallRyeGraphQLExecutionHandler extends SmallRyeGraphQLAbstractHand
     private static final String DEFAULT_REQUEST_CONTENT_TYPE = "application/json; charset="
             + StandardCharsets.UTF_8.name();
     private static final String MISSING_OPERATION = "Missing operation body";
+
+    private static final Logger log = Logger.getLogger(SmallRyeGraphQLExecutionHandler.class);
 
     public SmallRyeGraphQLExecutionHandler(boolean allowGet, boolean allowPostWithQueryParameters, boolean runBlocking,
             CurrentIdentityAssociation currentIdentityAssociation,
@@ -320,8 +325,13 @@ public class SmallRyeGraphQLExecutionHandler extends SmallRyeGraphQLAbstractHand
 
         @Override
         public void write(ExecutionResponse er) {
-
             if (shouldFail(er)) {
+                er.getExecutionResult()
+                        .getErrors().stream()
+                        .filter(e -> e.getErrorType().equals(ErrorType.ExecutionAborted))
+                        .forEach(e -> {
+                            log.error("Execution aborted", (AbortExecutionException) e);
+                        });
                 response.setStatusCode(500)
                         .end();
             } else {

